src/main/java/com/chain4travel/cmbplugin/grpc/services/ValidationServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+
+
+package com.chain4travel.cmbplugin.grpc.services;
+
+
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import com.chain4travel.cmbplugin.cache.CacheSearchType;
+import com.chain4travel.cmbplugin.cache.CacheService;
+import com.chain4travel.cmbplugin.grpc.converter.MessageConverter;
+
+import build.buf.gen.cmp.services.accommodation.v1alpha.AccommodationSearchRequest;
+import build.buf.gen.cmp.services.accommodation.v1alpha.AccommodationSearchResponse;
+import build.buf.gen.cmp.services.book.v1alpha.ValidationRequest;
+import build.buf.gen.cmp.services.book.v1alpha.ValidationResponse;
+import build.buf.gen.cmp.services.book.v1alpha.ValidationServiceGrpc.ValidationServiceImplBase;
+import io.grpc.stub.StreamObserver;
+import net.devh.boot.grpc.server.service.GrpcService;
+
+
+@GrpcService
+public class ValidationServiceImpl extends ValidationServiceImplBase {
+
+    private final static Logger logger = LoggerFactory.getLogger(ValidationServiceImpl.class);
+
+    @Autowired
+    private CacheService        cacheService;
+
+    @Value("${cmbplugin.cache.enabled}")
+    private boolean             cacheEnabled;
+
+
+    @Override
+    public void validation(ValidationRequest request, StreamObserver<ValidationResponse> responseObserver) {
+        var searchId = UUID.fromString(request.getSearchId());
+
+        // If legacy system is not stateful, get data from the cache and then validate
+        if (cacheEnabled && cacheService.getCacheSearchType(searchId) == CacheSearchType.accommodation) {
+            var cachedRequest = MessageConverter.readAccommodationSearchRequest(cacheService.readSearchRequestData(searchId));
+            var cachedResponse = MessageConverter.readAccommodationSearchResponse(cacheService.readSearchResponseData(searchId));
+
+            logger.debug("cachedRequest:\n{}", cachedRequest);
```

Comment:
Guess that is leftover debug code? Is that something we want to keep?

Reply:
I would keep it and since it is on debug level log, it does not hurt.